### PR TITLE
fix Issue 15208 - Eradicate all uses of "Enforcement failed" in Phobos

### PR DESCRIPTION
**Pull request by CyberShadow**
https://issues.dlang.org/show_bug.cgi?id=15208

### CODE REVIEW

File: std/concurrency.d [hunk 1]
```diff
@@ -2322,7 +2322,7 @@ private
         {
             assert( m_count );
             Node* n = r.m_prev;
-            enforce( n && n.next );
+            enforce( n && n.next, "attempting to remove invalid list node" );
```
Comment (CyberShadow) [inline std/concurrency.d]: Not totally sure about these ones... but these look like they should be asserts anyway (these are inside methods inside a private data structure). @complexmath ?

**Comment by Hackerpilot** [comment]
Looks good to me.

**Comment by JakobOvrum** [comment]
LGTM, although `enforce` on `malloc` isn't meaningful. They can be trivially replaced with calls to functions from #3031, or `onOutOfMemoryError` can be called manually at each call-site.

**Comment by AndrejMitrovic** [comment]
LGTM. The malloc comments could be part of a separate pull.

**Comment by AndrejMitrovic** [comment]
Auto-merge toggled on

**Pull request merged**